### PR TITLE
Add AST serialization

### PR DESCRIPTION
**Pull request by eklimo**
- Create `Serializer` class for serialization of AST nodes
- Add Gson dependency
- Add `third_party` `gson-extras` dependency
- Add unit tests for serialization
- Add test data files
- Move `TranslateHelpers` to common `testing` package

### CODE REVIEW

**Comment by eklimo** [comment]
All Kotlin files have been reviewed. Please review the `BUILD` files and `third_party` structure

File: src/main/java/com/google/summit/serialization/Serializer.kt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+    gson = builder.create()
+  }
+
+  /** Serializes the given [node] into a JSON string. */
+  fun serialize(node: Node): String {
+    return gson.toJson(node)
+  }
+
+  /** Deserializes the given [JSON string][json] into a node of type [nodeType]. */
```
Comment (aaronhurst-google) [inline src/main/java/com/google/summit/serialization/Serializer.kt:127]: Is this the intended Kdoc formatting?

File: WORKSPACE [hunk 2]
```diff
@@ -87,3 +87,13 @@ http_archive(
...
+
+maven_jar(
+    name = "gson",
+    artifact = "com.google.code.gson:gson:2.9.0",
+)
+
+maven_jar(
+    name = "kotlin-reflect",
+    artifact = "org.jetbrains.kotlin:kotlin-reflect:1.7.0",
+)
```
Comment (aaronhurst-google) [inline WORKSPACE]: I'm not going to enforce this in code review going forward... not a good use of time...

But if you can setup your editor to prefer newlines at EOF, that would be preferable.

**Review by aaronhurst-google — APPROVED** [review]

File: src/main/java/com/google/summit/serialization/Serializer.kt (hunk 1 above)
Comment (eklimo) [reply src/main/java/com/google/summit/serialization/Serializer.kt:127]: This is what I intended. Could it be improved?

Comment (aaronhurst-google) [reply src/main/java/com/google/summit/serialization/Serializer.kt:127]: I think I misparsed it.  LGTM.

**Pull request merged**